code/lightna/lightna-redis/App/Storage/Redis.php

```diff
@@ -67,8 +205,66 @@ public function getList(array $keys): array
         return $return;
     }
 
+    /**
+     * @throws \RedisException
+     */
     public function clean(array $tags): void
     {
-        // TODO: Implement clean() method.
+        foreach ($tags as $tag) {
+            foreach ($this->getKeysForTag($tag) as $key) {
+                $this->unset($key);
+            }
+        }
+    }
+
+    /**
+     * Clean tags
+     *
+     * @param string $key
+     * @param array $tags
+     *
+     * @throws \RedisException
+     */
+    private function cleanTags(string $key, array $tags): void
+    {
+        // Iterate over each tag and remove the key from the tag set
+        foreach ($tags as $tag) {
+            $this->client->sRem($tag, $key);
+
+            // Check if the tag set is now empty and remove it if it is
+            if ($this->client->sCard($tag) == 0) {
+                $this->client->del($tag);
+            }
+        }
+    }
+
+    /**
+     * Get tags for a key
+     *
+     * @param string $key
+     *
+     * @return array
+     * @throws \RedisException
+     */
+    private function getTagsForKey(string $key): array
+    {
+        // Get string from redis
+        $tags = $this->client->hGet($key, self::FIELD_TAGS);
+
+        return $tags ? json_decode((string)$tags) : [];
+    }
+
+    /**
+     * Get keys for a tag
+     *
+     * @param string $tag
+     *
+     * @return array
+     * @throws \RedisException
+     */
```

Comment:
return type is declared, param type is declared, description equal to method name. Seems these docs just duplicates the declaration, lets clean them